src/kubectl.ts

```diff
@@ -147,43 +153,35 @@ async function kubectllInternalArgs(args: string[], namespace: string | undefine
 }
 
 export async function baseKubectlPath(): Promise<string> {
-	let result : FindBinaryResult = await findBinary('kubectl');
+	const result : FindBinaryResult = await findBinary('kubectl');
 	if (result && result.output && result.err === null) {
 		return result.output;
 	}
-	let bin = config.getActiveKubectlconfig();
+	const bin: string = config.getActiveKubectlconfig();
 	if (!bin) {
-		bin = 'kubectl';
-		return bin;
+		return 'kubectl';
 	}
-	let binpath = path.normalize(bin);
-	return binpath;
-}
-
-interface FindBinaryResult {
-	err: number | null;
-	output: string;
+	return path.normalize(bin);
 }
 
 export async function findBinary(binName: string): Promise<FindBinaryResult> {
-	let cmd = `which ${binName}`;
+	let cmd: string = `which ${binName}`;
 
 	if (shell.isWindows()) {
 		cmd = `where.exe ${binName}.exe`;
 	}
 
-	const opts = {
+	const opts: any = {
 		async: true,
 		env: {
 			HOME: process.env.HOME,
 			PATH: process.env.PATH
 		}
 	};
 
-	const execResult = await shell.execCore(cmd, opts);
-	if (execResult.code) {
-		return { err: execResult.code, output: execResult.stderr };
+	const execResult: ShellResult = await shell.execCore(cmd, opts);
```

Comment:
tbh I am still not 100% sure I understand the original if clause. Imo a result code of `0` is signaling a normal execution of the process while other codes indicate issues. So the question for me would be why any code is handled as an "error" case instead of just the ones that return a code !== 0.

Reply:
Ok. I learned now that 0 will result in a falsy statement, so the original code might be fine. I still would keep the change for a better readability.